### ISSUE TITLE
[DynamoDB] Marshaler & UnMarshaler data for DynamoDB

### ISSUE THREAD

**Issue by tlfbrito** [async-aws/aws#707]
This is not a bug report.

I understand that this library is implements only the relevant, the most used operations of the AWS SDK, which is a great idea. But most of the times I feel the need of having a Marshaler to convert an array or a JSON document into the format required by DynamoDB and vice versa. Should this be part of this library? or do you think that should be a handle by another library
How do you serialize and unserialize data without this helper?

**Comment by tyx** [comment]
Yes, you're right ! We should find a way to include the AWS Marshaler in this lib. I guess manifest.json could be extended.

For now I map on my own my json document. Hopefully, I don't have many cases :smile:

**Comment by jderusse** [comment]
I'm not sure this should be part of the `async-aws/dynamo-db` package.
but a `async-aws/simple-dynamo-db` could provide such behabior (like we did fo S3: https://async-aws.com/integration/simple-s3.html)

**Comment by Nyholm** [comment]
Excellent idea. 

Yeah, see if creating a `simple-dynamo-db` library makes sense.

**Comment by tyx** [comment]
Hi guys,

I could have some time (and because I need it now :p) to work on this simple-dynamo-db. The question is around the Marshaler itself. Should we build a new one or should we use AWS one ?

If we want to use AWS one (I'm :+1: to use it), how do you think we could "retrieve" their code ? Like I said, through manifest or you have other idea ?

**Comment by tyx** [comment]
I just give some tries and the `unmarshalValue` method would require a lot of change. I guess we better make a new one

**Comment by Nyholm** [comment]
Hm.. 

Is this a blocker for a 1.0.0 release of dynamo DB client?

**Comment by Nyholm** [comment]
I'll release 1.0 of dynamodb now then,
